From 4986cf02e0d1094a68b9160de8d462ff3e5e55f6 Mon Sep 17 00:00:00 2001
From: Elijah Shakkour <elijahsh@mellanox.com>
Date: Mon, 15 Jul 2019 17:14:15 +0300
Subject: [PATCH] mlx5: Set MLX5 to be multifunction device

Issue: 1824873

This will set multifunction bit in config space header-type for PF.

According to PCI spec, this bit should be set to indicate
that device supports multifunction.

Change-Id: Ic5174af0f27f58fe31f172a806e461b635391316
Signed-off-by: Elijah Shakkour <elijahsh@mellanox.com>
---
 hw/net/mellanox/mlx5/mlx5.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/hw/net/mellanox/mlx5/mlx5.c b/hw/net/mellanox/mlx5/mlx5.c
index 6699ae23f..daf7b100e 100644
--- a/hw/net/mellanox/mlx5/mlx5.c
+++ b/hw/net/mellanox/mlx5/mlx5.c
@@ -883,7 +883,7 @@ static void mlx5_instance_init(Object *obj)
 {
     MLX5State *s = MLX5(obj);
 
-    PCI_DEVICE(s)->cap_present |= QEMU_PCI_CAP_EXPRESS;
+    PCI_DEVICE(s)->cap_present |= QEMU_PCI_CAP_EXPRESS | QEMU_PCI_CAP_MULTIFUNCTION;
 
     device_add_bootindex_property(obj, &s->conf.bootindex,
                                   "bootindex", NULL,
-- 
2.21.0

